mllib/src/main/scala/org/apache/spark/mllib/evaluation/MultilabelMetrics.scala

```diff
@@ -38,76 +40,62 @@ class MultilabelMetrics @Since("1.2.0") (predictionAndLabels: RDD[(Array[Double]
       (r.getSeq[Double](0).toArray, r.getSeq[Double](1).toArray)
     })
 
-  private lazy val numDocs: Long = predictionAndLabels.count()
+  /**
+   * Use MultilabelSummarizer to calculate all summary statistics of predictions
+   * and labels on one pass.
+   */
+  private lazy val summary: MultilabelSummary = {
+    predictionAndLabels
+      .treeAggregate(new MultilabelSummarizer)(
+        (summary, sample) => summary.add(sample._1, sample._2),
```

Comment:
Can `sample` be written as `(label, prediction)` for clarity? I forget whether Scala allows nested tuples here

Reply:
If I convert this to `(summary, (preds, l)) => summary.add(preds, l),` then the IDEA complains

Reply:
With a `{ case (...) =>` statement? OK no big deal